ProjectLighthouse.Servers.Website/Pages/UserPage.cshtml.cs

```diff
@@ -88,11 +90,15 @@ public async Task<IActionResult> OnGet([FromRoute] int userId)
         this.CommentsEnabled = ServerConfiguration.Instance.UserGeneratedContentLimits.LevelCommentsEnabled && this.ProfileUser.CommentsEnabled;
         if (this.CommentsEnabled)
         {
-            this.Comments = await this.Database.Comments.Include(p => p.Poster)
-                .OrderByDescending(p => p.Timestamp)
-                .Where(p => p.TargetId == userId && p.Type == CommentType.Profile)
-                .Take(50)
-                .ToListAsync();
+            this.Comments = await
+                (from comment in this.Database.Comments
+                where !(
+                    from blockedProfile in this.Database.BlockedProfiles 
+                    where blockedProfile.UserId == this.User.UserId 
```

Comment:
`this.User` can be null (does this work if you aren't logged in) so you might be able to do `this.User?.UserId`. This also looks kind of wacky but if it works and doesn't pull in the entire comment/blocked user table I'm fine with it